### PR TITLE
Added channel_rpc_timeout as possible option

### PR DESCRIPTION
**Pull request by bravoman**
Added  channel_rpc_timeout as a possible option for configuring the bundle default connection.

### CODE REVIEW

**Comment by bravoman** [comment]
Depends on this pull request: https://github.com/php-amqplib/php-amqplib/pull/1041

**Comment by bravoman** [comment]
https://github.com/php-amqplib/php-amqplib/pull/1041 is merged already. So now I guess a version bump on the php-amqplib side the composer requirement needs to be upgraded.

**Comment by Eloar** [comment]
It seems tests should be updated with new parameter.

**Comment by bravoman** [comment]
[mihaileu](https://github.com/mihaileu) is this anywhere near a solution? Or do I need some insight in the inner workings of this bundle? I'm quite new here. 

Oh and I guess we're still waiting for php-amqplib 3.4.1 here:
image

**Comment by mszczesniak94** [comment]
@bravoman 👏 

I see there are missing changes in https://github.com/php-amqplib/RabbitMqBundle/blob/master/DependencyInjection/Configuration.php
```
->scalarNode('channel_rpc_timeout')->defaultValue(0.0)->end()
```

It would be nice to add information about this parameter to readme eg. this feature requires `php-amqplib/php-amqplib: ^3.5.0` [change log here](https://github.com/php-amqplib/php-amqplib/blob/master/CHANGELOG.md) and add `channel_rpc_timeout` to listed parameters.

https://github.com/php-amqplib/RabbitMqBundle/blob/master/README.md

**Comment by mihaileu** [comment]
@bravoman an update in readme will help to move it forward

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -92,6 +92,7 @@ protected function addConnections(ArrayNodeDefinition $node)
                             ->booleanNode('lazy')->defaultFalse()->end()
                             ->scalarNode('connection_timeout')->defaultValue(3)->end()
                             ->scalarNode('read_write_timeout')->defaultValue(3)->end()
+                            ->scalarNode('channel_rpc_timeout')->defaultValue(0)->end()
```
Comment (bravoman) [inline DependencyInjection/Configuration.php:95]:
```suggestion
                            ->scalarNode('channel_rpc_timeout')->defaultValue(0.0)->end()
```

File: README.md [hunk 2]
```diff
@@ -113,6 +113,9 @@ old_sound_rabbit_mq:
             lazy:     false
             connection_timeout: 3
             read_write_timeout: 3
+            
+            # the timeout when waiting for a response from rabbitMQ (0 means waits forever)
+            channel_rpc_timeout: 0
```
Comment (bravoman) [inline README.md:118]:
```suggestion
            channel_rpc_timeout: 0.0
```

File: Tests/RabbitMq/AMQPConnectionFactoryTest.php [hunk 3]
```diff
@@ -351,6 +352,7 @@ public function testClusterConnectionParametersWithoutRootConnectionKeys()
                     'ssl_context'        => null,
                     'keepalive'          => false,
                     'heartbeat'          => 0,
+                    'channel_rpc_timeout' => 0,
```
Comment (bravoman) [inline Tests/RabbitMq/AMQPConnectionFactoryTest.php:355]:
```suggestion
                    'channel_rpc_timeout' => 0.0,
```

File: Tests/RabbitMq/AMQPConnectionFactoryTest.php [hunk 4]
```diff
@@ -500,6 +503,7 @@ public function testSSLClusterConnectionParameters()
                 'context' => null,
                 'keepalive' => false,
                 'heartbeat' => 0,
+                'channel_rpc_timeout' => 0,
```
Comment (bravoman) [inline Tests/RabbitMq/AMQPConnectionFactoryTest.php:506]:
```suggestion
                'channel_rpc_timeout' => 0.0,
```

File: Tests/RabbitMq/AMQPConnectionFactoryTest.php [hunk 5]
```diff
@@ -611,6 +616,7 @@ public function testSocketClusterConnectionParameters()
                     'read_write_timeout' => 3,
                     'read_timeout'       => 3,
                     'write_timeout'      => 3,
+                    'channel_rpc_timeout' => 0,
```
Comment (bravoman) [inline Tests/RabbitMq/AMQPConnectionFactoryTest.php:619]:
```suggestion
                    'channel_rpc_timeout' => 0.0,
```

File: DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -101,6 +102,7 @@ protected function addConnections(ArrayNodeDefinition $node)
                             ->booleanNode('keepalive')->defaultFalse()->info('requires php-amqplib v2.4.1+ and PHP5.4+')->end()
                             ->scalarNode('heartbeat')->defaultValue(0)->info('requires php-amqplib v2.4.1+')->end()
                             ->scalarNode('connection_parameters_provider')->end()
+
```
Comment (bravoman) [inline DependencyInjection/Configuration.php:105]:
```suggestion
```

**Review by bravoman — COMMENTED** [review]
Suggestion comments added

**Comment by bravoman** [comment]
Anything else needed for this to move forward?

**Pull request merged**